enterprise/cmd/frontend/authz/authz.go

```
@@ -106,10 +106,10 @@ func Init(d dbutil.DB, clock func() time.Time) {
 	})
 
 	// Enforce the use of a valid license key by preventing all HTTP requests if the license is invalid
-	// (due to a error in parsing or verification, or because the license has expired).
+	// (due to an error in parsing or verification, or because the license has expired).
 	hooks.PostAuthMiddleware = func(next http.Handler) http.Handler {
 		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-			if err := backend.CheckCurrentUserIsSiteAdmin(r.Context()); err != nil {
+			if err := backend.CheckCurrentUserIsSiteAdmin(r.Context()); err == nil {
```

`err == nil` indicates the user is site admin, not the other way around.

---

We want only site-admin to be exempt from license enforcement (based on the comment one-line down). Docstring of `backend.CheckCurrentUserIsSiteAdmin` states non-nil error indicate non-admin:

https://github.com/sourcegraph/sourcegraph/blob/4dac2900c47ba8bc785ea30e0d1808a6010e515d/cmd/frontend/backend/site_admin.go#L16-L17

---

https://github.com/sourcegraph/sourcegraph/blob/4dac2900c47ba8bc785ea30e0d1808a6010e515d/cmd/frontend/backend/site_admin.go#L29

---

Yeah, what is practical meaning of checking this specific error?

Users are able to sign in/sign up and get blocked with a license expired notice page right away?

---

I don't know, seems like good practice to check for the specific condition? What if we get a  database query error back instead?

---

Generally agree, but here we can only be sure the user is a site admin when there is absolutely no error. Any error could potentially mean the user is not a site admin (due to something not working and we can't get desired data back). 

So far we've been relying on `if err == nil` (or `if err != nil`) to indicate if the authenticated user is site admin almost everywhere in our codebase.

But yeah, we do swallow errors in this type of pattern. I'll add another `else if` to detect unexpected errors.

---

On a second thought, I think there is a practice meaning :)

Site admin should be able sign in (if not yet), so we should only block authenticated users.

---

New commits pushed, PTAL!

---

LGTM. Is this distinction something we should port over to other places of the codebase? If so, can you file a tech debt issue and  think of a good way to do so without a lot of code duplication? :D

---

> Is this distinction something we should port over to other places of the codebase? If so, can you file a tech debt issue and think of a good way to do so without a lot of code duplication? :D

After quick research, I think it's only necessary if we test for `err == nil` (here happens to be the only case). Otherwise, error is propagated to the caller therefore not swallowed.